holoviews/core/overlay.py

```diff
@@ -194,7 +194,7 @@ def dynamic_mul(*args, **kwargs):
             return other.clone(shared_data=False, callback=callback,
                                streams=[])
         elif not isinstance(other, ViewableElement):
-            return NotImplemented
+            raise NotImplementedError
```

Comment:
I remember there is special semantics of `NotImplemented` so now I am wondering if that was the right thing.

Reply:
I.e:

```suggestion
            return NotImplemented
```

Reply:
``return NotImplemented`` in ``__mul__`` force it to use ``__rmul__`` which is important to get the HoloMaps working.

Reply:
Right, I figured that out and fixed it accordingly. Was I right to remove the submodule reference update? The tests are now green.

Reply:
Yes, should be ready to merge.